### PR TITLE
Implement a try-to-dispatch concept

### PR DESCRIPTION
**Pull request by codemercenary**
This is useful for cases where consumers want to be able to retry lambdas from the dispatch queue while preserving order in single-threaded use cases.

### CODE REVIEW

File: src/autowiring/DispatchQueue.cpp [hunk 1]
```diff
@@ -67,6 +67,31 @@ void DispatchQueue::DispatchEventUnsafe(std::unique_lock<std::mutex>& lk) {
...
+  // deadlocks.
+  DispatchThunkBase* pThunk = m_pHead;
+  m_pHead = pThunk->m_pFlink;
+  lk.unlock();
+
+  try { (*pThunk)(); }
+  catch (...) {
+    // Failed to execute thunk, put it back
+    lk.lock();
+    pThunk->m_pFlink = m_pHead;
```
Comment (veronicaz41) [inline src/autowiring/DispatchQueue.cpp]: From reading the comments in the header, I thought we are putting this thunk back to the tail of the queue?

**Comment by veronicaz41** [comment]
Interesting. What is the use case for this?

File: src/autowiring/DispatchQueue.cpp (hunk 1 above)
Comment (codemercenary) [inline src/autowiring/DispatchQueue.cpp]: Negative, comment states that the item is replaced at the front of the queue on failure.

Comment (veronicaz41) [inline src/autowiring/DispatchQueue.cpp]: I see.

**Pull request merged**